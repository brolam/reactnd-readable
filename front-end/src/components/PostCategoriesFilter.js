import React from 'react'
import PropTypes from 'prop-types'

function PostCategoriesFilter(
  { categories,
    onSelected = e => { },
    selectedCategoryPath = 'none' }) {
  return (
    <select
      value={selectedCategoryPath}
      onChange={onSelected} >
      <option value="none">All Categories</option>
      {categories.map(category => (
        <option
          key={category.name}
          value={category.path} >{category.name[0].toUpperCase() + category.name.slice(1)}</option>
      ))}
    </select>
  )
}

PostCategoriesFilter.propTypes = {
  categories: PropTypes.array.isRequired
}

export default PostCategoriesFilter