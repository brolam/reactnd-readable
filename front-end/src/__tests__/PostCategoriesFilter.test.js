
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import PostCategoriesFilter from '../components/PostCategoriesFilter'

const categories = global.dataForTest.categories

test('render without error', () => {
  ReactDOM.render(<PostCategoriesFilter categories={categories} />, document.createElement('div'));
})

test('on selected a category', () => {
  const testOnSelectOneCategory = (category) => {
    const onSelectCategory = (e) => {
      const selectedCategory = e.target.value;
      expect(category).toEqual(selectedCategory);
    }
    const postCategoriesFilter = mount(
      <PostCategoriesFilter
        categories={categories}
        onSelected={onSelectCategory}
      />);
    const event = { target: { value: category } };
    postCategoriesFilter.find('select').simulate('change', event);
  };
  testOnSelectOneCategory("udacity");
  testOnSelectOneCategory("reactd");
  testOnSelectOneCategory("redux");
})

test('last Snapshot', () => {
  const postCategoriesFilter = renderer.create(
    <PostCategoriesFilter
      categories={categories}
      onSelected={e =>{}}
    />);
  let tree = postCategoriesFilter.toJSON();
  expect(tree).toMatchSnapshot();
})