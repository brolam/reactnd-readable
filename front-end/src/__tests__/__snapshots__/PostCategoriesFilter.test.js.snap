// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
<select
  onChange={[Function]}
>
  <option
    value="none"
  >
    All Categories
  </option>
  <option
    value="udacity"
  >
    Udacity
  </option>
  <option
    value="react"
  >
    React
  </option>
  <option
    value="redux"
  >
    Redux
  </option>
</select>
`;
